### PR TITLE
Added a job to push Origin releases to Docker Hub

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@smarterclayton PTAL

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

File: config/test_cases/push_origin_release.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      title: "transfer docker config to remote host"
+      script: |-
+        scp -F ./.config/origin-ci-tool/inventory/.ssh_config /var/lib/jenkins/.dockercfg openshiftdevel:/tmp/.dockercfg
+    - type: "script"
+      title: "push the release"
+      repository: "origin"
+      script: |-
+        sudo sed -i "s/OPTIONS='/OPTIONS='--skip-schema2-push /" /etc/sysconfig/docker
+        sudo systemctl restart docker
+        DOCKER_CONFIG=/tmp/.dockercfg OS_PUSH_BASE_REGISTRY="docker.io/" hack/push-release.sh
```
Comment (stevekuznetsov) [inline config/test_cases/push_origin_release.yml]: @danmcp you may know -- we aren't pushing the base image here, do we know why?

File: config/test_cases/push_origin_release.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  actions:
+    - type: "host_script"
+      title: "transfer docker config to remote host"
+      script: |-
+        scp -F ./.config/origin-ci-tool/inventory/.ssh_config /var/lib/jenkins/.dockercfg openshiftdevel:/tmp/.dockercfg
+    - type: "script"
+      title: "push the release"
+      repository: "origin"
+      script: |-
+        sudo sed -i "s/OPTIONS='/OPTIONS='--skip-schema2-push /" /etc/sysconfig/docker
```
Comment (stevekuznetsov) [inline config/test_cases/push_origin_release.yml]: @danmcp @smarterclayton do we have a timeline for migrating to schema 2?

File: config/test_cases/push_origin_release.yml (hunk 1 above)
Comment (danmcp) [inline config/test_cases/push_origin_release.yml]: @stevekuznetsov Not sure.

File: config/test_cases/push_origin_release.yml (hunk 2 above)
Comment (danmcp) [inline config/test_cases/push_origin_release.yml]: 3.6

Comment (stevekuznetsov) [inline config/test_cases/push_origin_release.yml]: So, as this pushing Origin images, we can drop this today, right?

File: config/test_cases/push_origin_release.yml (hunk 1 above)
Comment (smarterclayton) [inline config/test_cases/push_origin_release.yml]: It's built from the hub today

File: config/test_cases/push_origin_release.yml (hunk 2 above)
Comment (smarterclayton) [inline config/test_cases/push_origin_release.yml]: I stopped pushing images with schema1, we can switch

**Comment by smarterclayton** [comment]
Changes look fine to me otherwise

**Comment by stevekuznetsov** [comment]
Removed Schema 1 push, first job running: https://ci.openshift.redhat.com/jenkins/job/push_origin_release/1/console

**Comment by stevekuznetsov** [comment]
@smarterclayton looks like the credentials we have on the Jenkins master are for `https://index.docker.io/v1/` only

**Pull request closed without merging**